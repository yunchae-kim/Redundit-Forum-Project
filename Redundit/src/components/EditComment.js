/* eslint-disable no-param-reassign */
import { useState } from "react";
import "../style/components/EditComment.css";
import {
  Button, InputGroup,
} from "@blueprintjs/core";
import { useHistory } from "react-router-dom";
import { updateComment } from "../utility/axiosComment";
import { addHashtag, deleteHashtagByComment } from "../utility/axiosHashtags";

const EditComment = (props) => {
  const { location: { state: { comment, hashtags } } } = props;
  const [content, setContent] = useState(comment.content);
  const [tags, setTags] = useState(hashtags.map((item) => item.hashtag));
  const history = useHistory();

  const addTags = (event) => {
    if (event.key === "Enter" && event.target.value !== "") {
      if (!tags.includes(event.target.value)) {
        setTags([...tags, event.target.value]);
      }
      event.target.value = "";
    }
  };

  const removeTag = (targetTag) => {
    setTags([...tags.filter((tag) => tag !== targetTag)]);
  };

  const submitComment = () => {
    updateComment(comment.idcomment, content, true, false, "", "", "").then(() => {
      deleteHashtagByComment(comment.idcomment).then(() => {
        addHashtag(tags, comment.idcomment, "comment").then(() => {
          history.push(`/post/${comment.idpost}`, { postId: comment.idpost });
        });
      });
    });
  };

  return (
    <div className="Cmnt-">
      <div id="Cmnt-input">
        <InputGroup
          value={content}
          id="Cmnt-inputUser"
          placeholder="Input comment"
          large="true"
          type="text"
          onChange={(e) => setContent(e.target.value)}
        />
      </div>
      <div id="NP-tagInput">
        <ul id="NP-tags">
          {tags.map((tag) => (
            <li key={tag} className="tag">
              <span className="NP-tagTitle">
                {" "}
                {tag}
                {" "}
              </span>
              <Button
                className="NP-tagCloseIcon"
                icon="cross"
                minimal
                onClick={() => removeTag(tag)}
              />
            </li>
          ))}
        </ul>
        <input
          type="text"
          onKeyUp={(event) => addTags(event)}
          placeholder="Enter to add hashtags"
        />
      </div>
      <div className="NP-postBtnDiv">
        {content === "" ? <></> : (
          <Button
            id="NP-postBtn"
            text="Confirm"
            onClick={() => submitComment()}
          />
        )}
      </div>
    </div>
  );
};

export default EditComment;
