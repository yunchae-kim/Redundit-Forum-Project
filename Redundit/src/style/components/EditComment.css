.EC{
    position: fixed;
    z-index: 100;
    left: 25%;
    top: 30%;

    width: 50%;

    border: 1px solid #011627;
    border-radius: 1em;

    background-color: white;

    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;

    color: #011627;
}