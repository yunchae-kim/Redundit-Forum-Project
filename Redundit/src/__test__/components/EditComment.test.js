import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import EditComment from "../../components/EditComment";

const testComment = {
  iduser: 1,
  idpost: 1,
  content: "this is a new comment",
  status: false,
  flagged: false,
  image: "https://source.unsplash.com/random",
  audio: "https://source.unsplash.com/random",
  video: "https://source.unsplash.com/random",
};

const testHashTags = [{ hashtag: "tag1" }, { hashtag: "tag2" }];

const testProps = { state: { comment: testComment, hashtags: testHashTags } };

test("test should render EditComment component", async () => {
  render(
    <EditComment location={testProps} />,
  );
  const tag = screen.getByText("tag2");
  await waitFor(() => {
    expect(tag).toBeInTheDocument();
  });
});

test("test should EditComment match snapshot", () => {
  const tree = renderer.create(
    <EditComment location={testProps} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
