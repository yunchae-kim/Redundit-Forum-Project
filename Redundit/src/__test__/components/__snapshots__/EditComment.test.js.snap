// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should EditComment match snapshot 1`] = `
<div
  className="Cmnt-"
>
  <div
    id="Cmnt-input"
  >
    <div
      className="bp3-input-group bp3-large"
    >
      <input
        className="bp3-input"
        id="Cmnt-inputUser"
        onChange={[Function]}
        placeholder="Input comment"
        style={
          Object {
            "paddingLeft": undefined,
            "paddingRight": undefined,
          }
        }
        type="text"
        value="this is a new comment"
      />
    </div>
  </div>
  <div
    id="NP-tagInput"
  >
    <ul
      id="NP-tags"
    >
      <li
        className="tag"
      >
        <span
          className="NP-tagTitle"
        >
           
          tag1
           
        </span>
        <button
          className="bp3-button bp3-minimal NP-tagCloseIcon"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="bp3-icon bp3-icon-cross"
            icon="cross"
          >
            <svg
              data-icon="cross"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        className="tag"
      >
        <span
          className="NP-tagTitle"
        >
           
          tag2
           
        </span>
        <button
          className="bp3-button bp3-minimal NP-tagCloseIcon"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="bp3-icon bp3-icon-cross"
            icon="cross"
          >
            <svg
              data-icon="cross"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </li>
    </ul>
    <input
      onKeyUp={[Function]}
      placeholder="Enter to add hashtags"
      type="text"
    />
  </div>
  <div
    className="NP-postBtnDiv"
  >
    <button
      className="bp3-button"
      id="NP-postBtn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <span
        className="bp3-button-text"
      >
        Confirm
      </span>
    </button>
  </div>
</div>
`;
